codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -375,13 +381,51 @@ public void write(int allowedBytes) {
                         writePromise = ctx.newPromise();
                         writePromise.addListener(this);
                     }
+                    if (toWrite instanceof SlicedByteBuf && data instanceof CompositeByteBuf) {
```

Comment:
Can you just expand the comment a bit more to explain why all of this is needed?